format/leveldb/leveldb_table.go

```diff
@@ -441,3 +446,21 @@ func mask(crc uint32) uint32 {
 	// Rotate right by 15 bits and add a constant.
 	return ((crc >> 15) | (crc << 17)) + kMaskDelta
 }
+
+// Concatinate byteslices and convert into a string.
+func stringify(byteSlices ...[]byte) string {
+	var result []byte
+	for _, b := range byteSlices {
+		result = append(result, b...)
+	}
+	return string(result)
+}
+
+func fieldUTF8ReturnBytes(name string, nBytes int, d *decode.D) []byte {
```

Comment:
I've experiment and thought a bit about doing a decode api v2 that would use method chaining then things could probably be nicer, something like:
```
d.FieldUTF8("bla", nBytes).Bytes()
d.FieldUTF8("bla", nBytes).BitReader()
d.FieldUTF8("bla", nBytes).Map(...).Bytes()
```
Think that would make it a bit more flexible, would probably also cut down a bit on amount of generated code. Also! it can probably help make the api more type safe as ex `d.FieldUTF8` would return some type forcing things to use string. Should think more about this :)